src/parser/jobs/blm/modules/Gauge.js

```diff
@@ -154,19 +156,21 @@ export default class Gauge extends Module {
 	}
 
 	onGainPolyglot() {
-		if (this._hasPolyglot) {
-			this._overwrittenFoul++
+		this._polyglotStacks++
+		if (this._polyglotStacks > MAX_POLYGLOT_STACKS) {
+			this._overwrittenPolyglot++
 		}
-		this._hasPolyglot = true
+		this._polyglotStacks = Math.max(this._polyglotStacks, MAX_POLYGLOT_STACKS)
 		this.addEvent()
 	}
 
 	onConsumePolyglot() {
-		if (!this._hasPolyglot && this._overwrittenFoul > 0) {
+		if (!this._polyglotStacks > 0 && this._overwrittenPolyglot > 0) {
 			// Safety to catch ordering issues where Foul is used late enough to trigger our overwrite check but happens before Poly actually overwrites
-			this._overwrittenFoul--
+			this._overwrittenPolyglot--
 		}
-		this._hasPolyglot = false
+		this._polyglotStacks--
+		this._polyglotStacks = Math.max(this._polyglotStacks, 0)
```

Comment:
would generally prefer this be in one line, a-la `this.foo = Math.max(this.foo - 1, 0)`

Reply:
Fine by me, done.